cmd/asset_generation/discover/cloud_foundry/discover_test.go

```diff
@@ -0,0 +1,235 @@
+package cloud_foundry
+
+import (
+	"bufio"
+	"bytes"
+	"os"
+	"path/filepath"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+)
+
+var _ = Describe("Discover Manifest", func() {
+	var (
+		out          bytes.Buffer
+		writer       *bufio.Writer = bufio.NewWriter(&out)
+		tempDir      string
+		manifestPath string
+		// useLive bool
+	)
+
+	BeforeEach(func() {
+		tempDir, err := os.MkdirTemp("", "cloud_foundry_test")
+		Expect(err).NotTo(HaveOccurred())
+		manifestPath = filepath.Join(tempDir, "manifest.yaml")
+		input = manifestPath
+		output = ""
+		// Reset buffers before each test
+		writer.Reset(&out)
+	})
+
+	AfterEach(func() {
+		os.RemoveAll(tempDir)
+	})
+
+	DescribeTable("Manifest validation",
+		func(manifestContent string, expectedErrorMessage string) {
+			err := helperCreateTestManifest(manifestPath, manifestContent, 0644)
+			Expect(err).ToNot(HaveOccurred(), "Unable to create manifest.yaml")
+			input = manifestPath
+			output = ""
+			err = discoverManifest(writer)
+			writer.Flush()
+			if expectedErrorMessage != "" {
+				Expect(err).ToNot(BeNil(), "Expected an error due to invalid manifest content, got none")
+				Expect(err.Error()).To(ContainSubstring(expectedErrorMessage))
+			} else {
+				Expect(err).To(BeNil(), "Expected no error for invalid manifest, but got one")
```

Comment:
suggestion: another way to validate if an error has occured or not is with the `HaveOcurred()` function:
```
Expect(err).ToNot(HaveOccurred())
```
Just in case you'd like to use a more BDD approach.

Reply:
I was using some `HaveOccurred` and some `BeNil`, they were leftover.
Thanks for catching!